homeassistant/components/template/light.py

```diff
@@ -24,60 +26,136 @@
 
-from .const import DOMAIN
+from . import TriggerUpdateCoordinator
+from .const import CONF_OBJECT_ID, CONF_PICTURE, DOMAIN
 from .template_entity import (
+    LEGACY_FIELDS as TEMPLATE_ENTITY_LEGACY_FIELDS,
+    TEMPLATE_ENTITY_AVAILABILITY_SCHEMA,
     TEMPLATE_ENTITY_COMMON_SCHEMA_LEGACY,
+    TEMPLATE_ENTITY_ICON_SCHEMA,
     TemplateEntity,
     rewrite_common_legacy_to_modern_conf,
 )
+from .trigger_entity import TriggerEntity
 
 _LOGGER = logging.getLogger(__name__)
 _VALID_STATES = [STATE_ON, STATE_OFF, "true", "false"]
 
 # Legacy
+CONF_COLOR = "color"
 CONF_COLOR_ACTION = "set_color"
 CONF_COLOR_TEMPLATE = "color_template"
 
+CONF_HS = "hs"
 CONF_HS_ACTION = "set_hs"
 CONF_HS_TEMPLATE = "hs_template"
 CONF_RGB_ACTION = "set_rgb"
 CONF_RGB_TEMPLATE = "rgb_template"
+CONF_RGBW = "rgbw"
 CONF_RGBW_ACTION = "set_rgbw"
 CONF_RGBW_TEMPLATE = "rgbw_template"
+CONF_RGBWW = "rgbww"
 CONF_RGBWW_ACTION = "set_rgbww"
 CONF_RGBWW_TEMPLATE = "rgbww_template"
 CONF_EFFECT_ACTION = "set_effect"
+CONF_EFFECT_LIST = "effect_list"
 CONF_EFFECT_LIST_TEMPLATE = "effect_list_template"
 CONF_EFFECT_TEMPLATE = "effect_template"
+CONF_LEVEL = "level"
 CONF_LEVEL_ACTION = "set_level"
 CONF_LEVEL_TEMPLATE = "level_template"
+CONF_MAX_MIREDS = "max_mireds"
 CONF_MAX_MIREDS_TEMPLATE = "max_mireds_template"
+CONF_MIN_MIREDS = "min_mireds"
 CONF_MIN_MIREDS_TEMPLATE = "min_mireds_template"
 CONF_OFF_ACTION = "turn_off"
 CONF_ON_ACTION = "turn_on"
-CONF_SUPPORTS_TRANSITION = "supports_transition_template"
+CONF_SUPPORTS_TRANSITION = "supports_transition"
+CONF_SUPPORTS_TRANSITION_TEMPLATE = "supports_transition_template"
 CONF_TEMPERATURE_ACTION = "set_temperature"
+CONF_TEMPERATURE = "temperature"
 CONF_TEMPERATURE_TEMPLATE = "temperature_template"
 CONF_WHITE_VALUE_ACTION = "set_white_value"
+CONF_WHITE_VALUE = "white_value"
 CONF_WHITE_VALUE_TEMPLATE = "white_value_template"
 
-LIGHT_SCHEMA = vol.All(
+
+LEGACY_FIELDS = TEMPLATE_ENTITY_LEGACY_FIELDS | {
+    CONF_VALUE_TEMPLATE: CONF_STATE,
+    CONF_RGB_TEMPLATE: CONF_RGB,
+    CONF_RGBW_TEMPLATE: CONF_RGBW,
+    CONF_RGBWW_TEMPLATE: CONF_RGBWW,
+    CONF_EFFECT_LIST_TEMPLATE: CONF_EFFECT_LIST,
+    CONF_EFFECT_TEMPLATE: CONF_EFFECT,
+    CONF_LEVEL_TEMPLATE: CONF_LEVEL,
+    CONF_MAX_MIREDS_TEMPLATE: CONF_MAX_MIREDS,
+    CONF_MIN_MIREDS_TEMPLATE: CONF_MIN_MIREDS,
+    CONF_SUPPORTS_TRANSITION_TEMPLATE: CONF_SUPPORTS_TRANSITION,
+    CONF_TEMPERATURE_TEMPLATE: CONF_TEMPERATURE,
+    CONF_WHITE_VALUE_TEMPLATE: CONF_WHITE_VALUE,
+    CONF_COLOR_TEMPLATE: CONF_COLOR,
+    CONF_HS_TEMPLATE: CONF_HS,
+}
+
+DEFAULT_NAME = "Template Light"
+
+LIGHT_SCHEMA = (
+    vol.Schema(
+        {
+            vol.Exclusive(CONF_COLOR_ACTION, "hs_legacy_action"): cv.SCRIPT_SCHEMA,
+            vol.Exclusive(CONF_COLOR, "hs_legacy_template"): cv.template,
+            vol.Exclusive(CONF_HS_ACTION, "hs_legacy_action"): cv.SCRIPT_SCHEMA,
+            vol.Exclusive(CONF_HS, "hs_legacy_template"): cv.template,
```

Comment:
Please add comments which make it clear which one is the deprecated and which one is the modern configuration parameter. Also, I don't think we should have support for the deprecated configuration parameter in the modern schema. Instead we should just have support for hs action + hs template, and rename color action + color template in the legacy schema rewriter.